zppy/templates/tc_analysis.bash

```diff
@@ -90,6 +90,13 @@ cat ${result_dir}out.dat0* > ${result_dir}cyclones_${file_name}.txt
 StitchNodes --in_fmt "lon,lat,slp,wind" --in_connect ${result_dir}connect_CSne${res}_v2.dat --range 6.0 --mintime 6 --maxgap 1 --in ${result_dir}cyclones_${file_name}.txt --out ${result_dir}cyclones_stitch_${file_name}.dat --threshold "wind,>=,17.5,6;lat,<=,40.0,6;lat,>=,-40.0,6"
 rm ${result_dir}cyclones_${file_name}.txt
 
+# If cyclones_stitch file is empty, exit
+if ! [ -s ${result_dir}cyclones_stitch_${file_name}.dat ]; then
+  cd {{ scriptDir }}
+  echo 'ERROR (1)' > {{ prefix }}.status
+  exit 1
+fi
+
```

Comment:
Catches the error in https://github.com/E3SM-Project/e3sm_diags/pull/866#issuecomment-2405873119.

Reply:
@forsyth2 I suggest not to catch this error, so that `e3sm_diags` runs won't be blocked by empty stitch files.

Reply:
Based on your comment here: https://github.com/E3SM-Project/zppy/pull/632#issuecomment-2417430619.

Reply:
@chengzhuzhang I think it would be a mistake to let errors slide by silently, especially considering this one was hard to track down.

A couple options on the top of my mind:
1. Leave this as-is. If `tc_analysis` fails because of this, then users can remove `tc_analysis` from their list of sets in  the `e3sm_diags` task and then re-run `zppy`.
2. Have a parameter something like `fail_on_empty_stitch` to toggle this error on/off

Reply:
My take is that this is not an error, just no cyclone is detected. Without catching the error, it should work more smoothly to print in the log that indicate no cycle stitch file is generated.

Reply:
That makes sense, but it _does_ cause an error in E3SM Diags down the road then. Notably, the `IndexError: list index out of range` error on https://github.com/E3SM-Project/zppy/issues/625#issuecomment-2401000188 didn't add much clarity to the error.

I think if we remove this error check in the `tc_analysis` bash script (which I agree we should do if it's not actually an error per-se), we should add an empty-dat check in the `e3sm_diags` task.
I've copied the stack trace again here:
```
Traceback (most recent call last):
  File "/home/ac.forsyth2/miniconda3/envs/e3sm_diags_1003/lib/python3.10/site-packages/e3sm_diags/parameter/core_parameter.py", line 266, in _run_diag
    single_result = module.run_diag(self)
  File "/home/ac.forsyth2/miniconda3/envs/e3sm_diags_1003/lib/python3.10/site-packages/e3sm_diags/driver/tc_analysis_driver.py", line 91, in run_diag
    test_data["metrics"] = generate_tc_metrics_from_te_stitch_file(test_te_file)
  File "/home/ac.forsyth2/miniconda3/envs/e3sm_diags_1003/lib/python3.10/site-packages/e3sm_diags/driver/tc_analysis_driver.py", line 172, in generate_tc_metrics_from_te_stitch_file
    te_stitch_vars = _get_vars_from_te_stitch(lines, max_len, num_storms)
  File "/home/ac.forsyth2/miniconda3/envs/e3sm_diags_1003/lib/python3.10/site-packages/e3sm_diags/driver/tc_analysis_driver.py", line 249, in _get_vars_from_te_stitch
    year_start = int(lines[0].split("\t")[2])
IndexError: list index out of range
```
```
    test_te_file = os.path.join(
        test_data_path,
        "cyclones_stitch_{}_{}_{}.dat".format(test_name, test_start_yr, test_end_yr),
    )
```
So, it would be nice to add a non-empty check to that `test_te_file`. Based on the stack trace, it would probably be easier to add that in `e3sm_diags` than in `zppy`.

Reply:
Yes, it makes sense to catch it in e3sm_diags.

Reply:
Sounds good. And no, I don't think these `dat` files are used in any other sets. The TC analysis task isn't even a dependency for any other set:
```
    if "tc_analysis" in c["sets"]:
        dependencies.append(os.path.join(script_dir, f"tc_analysis{status_suffix}"))
```